### PR TITLE
Fix remove old pyds

### PR DESCRIPTION
**Pull request by poshul**
### **User description**
Address https://github.com/OpenMS/OpenMS/issues/7608
Now with the correct spelling.

___

### **PR Type**
bug_fix, configuration changes

___

### **Description**
- Fixed a typo in the `pyopenms-wheels.yml` workflow file by changing the file extension from `.so` to `.pyd`, ensuring the correct cleanup of old pyopenms libraries.
- Updated the cleanup command to remove obsolete pyopenms `.pyd` files, addressing issue #7608.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pyopenms-wheels.yml</strong><dd><code>Correct file extension for pyopenms cleanup in workflow</code>&nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/pyopenms-wheels.yml

<li>Fixed typo in file extension from <code>.so</code> to <code>.pyd</code>.<br> <li> Updated the cleanup command for pyopenms libraries.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7616/files#diff-084f20ae9727db0905c5fbbc32ef797e048c87887ccfc57b0f89b2bf82f45979">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: .github/workflows/pyopenms-wheels.yml [hunk 1]
```diff
@@ -135,7 +135,7 @@ jobs:
           conda remove --name "pyoms-bld-${pynodot}" --all
 
           # clean previous pyopenms libs
-          find . -name "pyopenms*.so" -exec rm -rf {} \;
+          find . -name "pyopenms*.pyd" -exec rm -rf {} \;
```
Comment (timosachsenberg) [inline .github/workflows/pyopenms-wheels.yml]: note that it is _pyopenms on windows

**Comment by jpfeuffer** [comment]
I think Linux and macOS wheels are also affected. (because of the underscore)

**Comment by poshul** [comment]
hmm, yup, looks like it does.
My linux test env is 3.9 so I'd missed that misbehavior.

**Comment by timosachsenberg** [comment]
awesome...
there seem to be some issue with brew though (unrelated?)

==> Installing ghostscript dependency: tesseract
==> Installing dependencies for tesseract: python@3.12 and harfbuzz
==> Installing tesseract dependency: python@3.12
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Error: The `brew link` step did not complete successfully
Could not symlink bin/2to3-3.12
Target /usr/local/bin/2to3-3.12
already exists. You may want to remove it:
  rm '/usr/local/bin/2to3-3.12'

To force the link and overwrite all conflicting files:
  brew link --overwrite python@3.12

To list all files that would be deleted:
  brew link --overwrite python@3.12 --dry-run

**Comment by poshul** [comment]
@timosachsenberg  this looks suspiciously like an encore of https://github.com/OpenMS/OpenMS/pull/7339

File: .github/workflows/pyopenms-wheels.yml [hunk 2]
```diff
@@ -300,7 +300,7 @@ jobs:
         ## Update the package lists for Brew
         brew update
         ## Needed for Qt. Install before to overwrite the default softlinks on the GH runners
-        brew install python3 --force --overwrite
+        brew install python@3.13 --force --overwrite
```
Comment (timosachsenberg) [inline .github/workflows/pyopenms-wheels.yml]: 3.13 correct? just saw 3.12 above without further looking into it

Comment (poshul) [reply .github/workflows/pyopenms-wheels.yml]: err, correct, it should be 3.12.

**Comment by poshul** [comment]
I've just checked the wheels this produces, and it looks like it fixes the problem.

**Pull request merged**